cmake/HPX_GeneratePackage.cmake

```diff
@@ -32,7 +32,7 @@ foreach(lib ${HPX_LIBRARIES})
     set(_is_debug FALSE)
     set(_is_release FALSE)
   else()
-    if(NOT EXISTS "${lib}")
+    if(NOT EXISTS "${lib}" AND NOT "${lib}" MATCHES "^-")
```

Comment:
If `${LIB}` does not exist, how can it start with a `-`?

Reply:
`EXISTS` checks if its argument exists in the filesystem (hooray CMake). The idea is (I guess) that if `${lib}` is a full path it goes in just like that, and otherwise it becomes `-l${lib}`. There seem to be cases though where `-pthread` or `-lpthread` are part of `HPX_LIBRARIES` which then become `-l-pthread` or `-l-lpthread`. I _think_ they come from `Boost_LIBRARIES` which can be different in different versions and on different platforms.

Reply:
Ok, sounds about right. Thanks!